app/controllers/subscriptions_controller.rb

```diff
@@ -68,36 +71,42 @@ def create
 
   def update
     authorize :subscription
+    # Overwrite the old subscription with the new values
     @subscription = subscription_params
     @subscription['EmailAddress'] = get_subscriber_email(@subscription['SubscriberId'])
-    native_id = params.permit(:native_id)['native_id']
 
-    subscription_response = cmr_client.ingest_subscription(@subscription.to_json, current_user.provider_id, native_id, token)
+    subscription_response = cmr_client.ingest_subscription(@subscription.to_json, current_user.provider_id, @native_id, token)
     if subscription_response.success?
       flash[:success] = I18n.t('controllers.subscriptions.update.flash.success')
-      redirect_to subscription_path(subscription_response.parsed_body['result']['concept_id'], subscription: @subscription, native_id: native_id)
+
+      # Hitting CMR too quickly in dev environment.
+      sleep(1) if Rails.env.development? || Rails.env.test?
```

Comment:
how did this come up? Why did we need to do this?

Reply:
I see from the demo why you are doing this, but I think we should find some other solution if possible. If no other solution exists then this idea is not horrible.

Reply:
addressed, will comment to explain below